courses/migrations/0013_backfill_paid_courserun.py

```diff
@@ -0,0 +1,31 @@
+# Generated by Django 3.2.14 on 2022-07-28 20:52
+
+from django.db import migrations
+
+
+def backfill_paidcourserun(apps, schema_editor):
+    """Backfill PaidCourseRun for existing orders"""
+    PaidCourseRun = apps.get_model("courses", "PaidCourseRun")
+    CourseRun = apps.get_model("courses", "CourseRun")
+    Order = apps.get_model("ecommerce", "Order")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    for order in Order.objects.filter(state__in=["fulfilled", "review"]):
+        # couldn't use order.purchased_runs here from app defined model
+        content_type = ContentType.objects.get_for_model(CourseRun)
+        for run_obj in order.lines.filter(purchased_content_type=content_type):
+            course_run = CourseRun.objects.get(pk=run_obj.purchased_object_id)
+            PaidCourseRun.objects.get_or_create(
+                order=order, course_run=course_run, user=order.purchaser
+            )
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("courses", "0012_paidcourserun"),
+    ]
+
+    operations = [
+        migrations.RunPython(backfill_paidcourserun),
```

Comment:
I tried it with or without migrations.RunPython.noop, It didn't seem to make much difference when I revert this migration. Could you explain more about the benefit of passing migrations.RunPython.noop here? sorry but I just wanted to understand the difference in case I miss anything

Reply:
By reverting a migration, I mean to go back to the previous migration. I am not sure why it isn't raising an exception in your case.
Here is a bit of my understanding:

1. There can be cases where you want to revert a migration like in this case we have a migration 0013. Let's say we have created and applied this migration without `noop`. Now, If we want to go back to a previous migration like 0012 then it will raise an exception that this operation is not reversible (As in Screenshot).

2. In the case of Data Migrations, we usually provide `noop` which means that we don't want to perform any action on the database while reverting a migration. We can also provide a callable in place of `noop` if we want to reverse our changes.
3. the last para in this guide explains it a bit. https://docs.djangoproject.com/en/3.2/topics/migrations/#data-migrations

<img width="1437" alt="Screenshot 2022-07-30 at 11 32 21 AM" src="https://user-images.githubusercontent.com/52656433/181878112-db9edf7c-f332-4424-b3a6-eebf6517e695.png">

Reply:
@asadali145 Thank you for the detail. Yeah I understand that migrations.RunPython.noop is just not doing anything if we want to revert. Without it, it would raise exception. In this case, once the migration is applied we shouldn't do anything as this is data migration as you mentioned, not something we can revert. In short, I agree that we should pass noop for good practice